salt/fileserver/roots.py

```diff
@@ -138,10 +138,16 @@ def serve_file(load, fnd):
     return ret
 
 
-def update():
+def update(**kwargs):
     """
     When we are asked to update (regular interval) lets reap the cache
     """
+    # __pub_user responsible for the runner can be passed but we don't do anything with it
+    if "__pub_user" in kwargs:
+        del kwargs["__pub_user"]
+    if kwargs:
+        raise ValueError("Unexpected keyword arguments received")
```

Comment:
It would be helpful to include `kwargs` in the exception. In the future, if something calls `update()`, it will be hard to `del` the argument from `kwargs` without knowing anything about it.